synapse/handlers/register.py

```diff
@@ -1013,11 +1013,11 @@ async def _register_email_threepid(
             user_tuple = await self.store.get_user_by_access_token(token)
             # The token better still exist.
             assert user_tuple
-            token_id = user_tuple.token_id
+            device_id = user_tuple.device_id
 
             await self.pusher_pool.add_or_update_pusher(
                 user_id=user_id,
-                access_token=token_id,
+                device_id=device_id,
```

Comment:
Shouldn't we not provide a `device_id` here? (Otherwise, why is it `Optional`?)

Reply:
I think the answer to this is that it is optional, but should almost always be provided. It seems the only time it *isn't* is when a user is registered via the admin API w/ an email.